### PR TITLE
[DNM] use rocksdb 5.8

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: librocksdb_sys/build.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ macro_rules! t {
 fn main() {
     let mut cfg = build_rocksdb();
 
-    cfg.cpp(true).file("crocksdb/c.cc");
+    cfg.cpp(true).flag_if_supported("-fno-rtti").file("crocksdb/c.cc");
```
Comment (siddontang) [inline librocksdb_sys/build.rs]: why do we need to check no-rtti here?

**Comment by BusyJay** [comment]
I thought we were going to use pingcap fork.

**Comment by huachaohuang** [comment]
We can use this branch https://github.com/pingcap/rocksdb/tree/pingcap/v5.8.

**Comment by siddontang** [comment]
Ping @zhangjinpeng1987

File: librocksdb_sys/build.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply librocksdb_sys/build.rs]: RocksDB abandon `dynamic_cast<T>`, we need to make consistency with RocksDB.

**Comment by siddontang** [comment]
PTAL @huachaohuang

**Comment by siddontang** [comment]
CI still failed @zhangjinpeng1987

**Comment by huachaohuang** [comment]
RocksDB 5.8 adds a `COMPACTION_OPTIMIZED_DEL_DROP_OBSOLETE`  ticker, breaks some of the `DBStatisticsTickerType` enums, please update them accordingly.
I think we may just list all of the tickers in the same order of RocksDB, instead of relying on the explicit enum values.

**Comment by siddontang** [comment]
any update?

**Pull request closed without merging**